### DISCUSSION TITLE
change widgets class depending on output of script

### DISCUSSION THREAD

**Discussion by justinesmithies** [elkowar/eww#681] [Q&A] [answered]
Is it possible to change the class on a widget depending if the output of that widget has "" or "some text inside" ?
The widget works but I'd like to see if I can change the class so it removes the padding-right if the value is = "" or empty.

```
(defwidget scratchpad-indicator []
  (box 
       :class "scratchpad"
       :orientation "h"
       :space-evenly false
       :halign "end"
       :spacing 10
    (button
        :class ""
	:onclick "scripts/scratchpad-indicator.sh ShowInfo"
        scratchpad)
))

(defpoll scratchpad :interval "1s"
  "scripts/scratchpad-indicator.sh Update")
```
 So if the value of scratchpad is an empty string or "" then change the line `:class "scratchpad"` to `:class "empty-scratchpad"`
Is this doable so it will change on the fly ????

**Comment by SwiftyChicken** [comment, marked as answer]
Yes this is doable but it's not straight forward as you would think. This because one of the side effects/ feature of defpoll is that it cannot set its variable to an empty string if it's already set to something that isn't an empty string. To work around this problem you will probably need to edit your code like so:
 ```
 (defwidget scratchpad-indicator []
  (box 
       :class "${matches(scratchpad, 'null')?'empty-scratchpad':'scratchpad'}"
       :orientation "h"
       :space-evenly false
       :halign "end"
       :spacing 10
    (button
        :class ""
	:onclick "scripts/scratchpad-indicator.sh ShowInfo"
        scratchpad)
))

(defpoll scratchpad :interval "1s"
  "scripts/scratchpad-indicator.sh Update|| echo null")
 ```
 but if you expect null to be one of the expected output you should change it accordingly.

**Reply by justinesmithies** [reply]
Thank you this worked but instead of `null` I added a regex to check for an empty string `^\s*$` and this works perfectly.
